src/backend/graphics/egl.rs

```diff
@@ -0,0 +1,662 @@
+//! Common traits and types for opengl rendering on graphics backends
+
+/// Large parts of the following file are taken from
+/// https://github.com/tomaka/glutin/tree/master/src/api/egl at commit
+/// `044e651edf67a2029eecc650dd42546af1501414`
+///
+/// It therefor falls under glutin's Apache 2.0 license
+/// (see https://github.com/tomaka/glutin/blob/master/LICENSE)
+
+use super::GraphicsBackend;
+
+use libloading::Library;
+use nix::{c_int, c_void};
+
+use std::ffi::{CStr, CString};
+use std::error::{self, Error};
+use std::fmt;
+use std::io;
+use std::ptr;
+use std::mem;
+
+mod ffi {
+    use nix::c_void;
+    use nix::libc::{uint64_t, int32_t, c_long};
+
+    pub type khronos_utime_nanoseconds_t = khronos_uint64_t;
+    pub type khronos_uint64_t = uint64_t;
+    pub type khronos_ssize_t = c_long;
+    pub type EGLint = int32_t;
+    pub type EGLNativeDisplayType = NativeDisplayType;
+    pub type EGLNativePixmapType = NativePixmapType;
+    pub type EGLNativeWindowType = NativeWindowType;
+    pub type NativeDisplayType = *const c_void;
+    pub type NativePixmapType = *const c_void;
+    pub type NativeWindowType = *const c_void;
+
+    pub mod egl {
+        use super::*;
+
+        include!(concat!(env!("OUT_DIR"), "/egl_bindings.rs"));
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum Native {
+    X11(ffi::NativeDisplayType, ffi::NativeWindowType),
+    Wayland(ffi::NativeDisplayType, ffi::NativeWindowType),
+    Gbm(ffi::NativeDisplayType, ffi::NativeWindowType),
```

Comment:
Yes. This was the main motivation of doing this change now, as this is essential to make anything useful out of the drm implementation I am working on.